### DISCUSSION TITLE
[SOLVED] Repository inheritance - does it work correctly?

### DISCUSSION THREAD

**Discussion by rembish** [devpi/devpi#1004] [General]
We have the next index structure:
```
company/stable
\---> root/pypi
\----> company/thirdparty
```
We've uploaded a `package==1.0.0+something` with our company patches to `company/thirdparty`.
It caused a problem that the clients who used `package==1.0.0` were not able to obtain it when asking it from `company/stable` index. Does it work as it should? Or we are missing some kind of configuration? Or is it a bug?

**Comment by fschulze** [comment]
Packages from mirrors are blocked by default if a private package with the same name is uploaded. This is to prevent supply chain attacks. See https://devpi.net/docs/devpi/devpi/stable/+doc/userman/devpi_indices.html?highlight=mirror_whitelist#modifying-the-mirror-whitelist

**Reply by rembish** [reply]
So, in this case we should whitelist (with intersection inheritance policy) `package` on `company/thirdparty` index, right?

**Reply by fschulze** [reply]
Yes

**Reply by rembish** [reply]
Thank you, Florian, you helped a lot.
